src/lib/core/components/EntityDiagram.tsx

```diff
@@ -0,0 +1,76 @@
+import { useStudyMetadata } from '../hooks/workspace';
+import { useEntityCounts } from '../hooks/entityCounts';
+import { SessionState } from '../../core';
+import MiniDiagram from '@veupathdb/components/lib/EntityDiagram/MiniDiagram';
+import ExpandedDiagram from '@veupathdb/components/lib/EntityDiagram/ExpandedDiagram';
+import {
+  ShadingData,
+  ShadingValue,
+} from '@veupathdb/components/lib/EntityDiagram/Types';
+import { StudyEntity } from '../types/study';
+import { VariableLink } from './VariableLink';
+
+interface Props {
+  sessionState: SessionState;
+  expanded: boolean;
+  orientation: 'horizontal' | 'vertical';
+  selectedEntity: string;
+}
+
+export function EntityDiagram(props: Props) {
+  const { sessionState, expanded, orientation, selectedEntity } = props;
+
+  const studyMetadata = useStudyMetadata();
+  const { session } = sessionState;
+  const { value: counts, error: countsError } = useEntityCounts();
+  const { value: filteredCounts, error: filteredCountsError } = useEntityCounts(
+    session?.filters
+  );
```

Comment:
I think we can pass these values in as props. We're already getting them in `Subsetting.tsx`. If we do that, then I believe we can remove the `sessionState` prop.

Can be a follow-up.